tools/rust-rofi_voxel_reconfig/rofi-voxel

```diff
@@ -1,33 +1,122 @@
-#!/usr/bin/env bash
+#!/usr/bin/env python3
 
-if [[ $# -ne 2 ]] ; then
-    >&2 echo "Expected 2 args: <input_world_file> <goal_world_file>"
-    exit 1
-fi
+import argparse
+import io
+import os
+import subprocess
+import sys
+import tempfile
+from datetime import datetime
+from typing import List, Optional
 
-if [ -z "$ROFI_BUILD_DIR" ]; then
-    >&2 echo "ROFI_BUILD_DIR is not set. Are you in RoFI environment?";
-    exit 1;
-fi
+LogFile = Optional[io.TextIOWrapper]
 
-TMP_DIR="$ROFI_BUILD_DIR/desktop/tmp"
-mkdir -p "$TMP_DIR" || exit 1
 
-INIT_VOXEL_FILE=$(mktemp "$TMP_DIR/tmp-init_voxel_world.XXXXXX.json")
-GOAL_VOXEL_FILE=$(mktemp "$TMP_DIR/tmp-goal_voxel_world.XXXXXX.json")
-RESULT_FILE=$(mktemp "$TMP_DIR/tmp-result_voxel_worlds.XXXXXX.json")
-trap "rm -f \"$INIT_VOXEL_FILE\" \"$GOAL_VOXEL_FILE\" \"$RESULT_FILE\"" EXIT INT QUIT TERM
+def log(s: str, logfile: LogFile):
+    print(f"[{datetime.now().strftime('%T.%f')} {os.path.basename(__file__)}]", s,
+          file=sys.stderr if logfile is None else logfile,
+          flush=True)
```

Comment:
Nechceš zvážit použít https://docs.python.org/3/library/logging.html?

Reply:
Použito